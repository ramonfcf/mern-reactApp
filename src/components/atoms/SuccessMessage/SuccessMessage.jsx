import propTypes from "prop-types";

const SuccessMessage = ({ message }) => {
  return (
    <div className="alert alert-success" role="alert">
      Success: {message}
    </div>
  );
};

export default SuccessMessage;

SuccessMessage.propTypes = {
  message: propTypes.string,
};
