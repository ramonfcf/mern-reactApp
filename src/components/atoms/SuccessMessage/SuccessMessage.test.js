import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import SuccessMessage from "./SuccessMessage";

describe("SuccessMessage component", () => {
  it("matches snapshot", () => {
    const { container } = render(<SuccessMessage message="Test message" />);
    expect(container).toMatchSnapshot();
  });

  it("renders with correct success message", () => {
    render(<SuccessMessage message="Test message" />);
    const successMessageElement = screen.getByRole("alert");

    expect(successMessageElement).toHaveTextContent("Success: Test message");
  });
});
