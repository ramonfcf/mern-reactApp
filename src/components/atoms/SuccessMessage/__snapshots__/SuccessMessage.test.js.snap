// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessMessage component matches snapshot 1`] = `
<div>
  <div
    class="alert alert-success"
    role="alert"
  >
    Success: 
    Test message
  </div>
</div>
`;
